usecase/zipcodes.go

```diff
@@ -0,0 +1,183 @@
+	fmt.Printf("Droppped table")
+	return true
+}
+
+// SearchZipCodes Search the zipcodes database the neighborhood by zip code
+func SearchZipCodes(cp string, collection *mongo.Collection) []models.ZipCodeCSV {
+	findOpts := options.Find()
+	var results []models.ZipCodeCSV
+	filter := bson.D{{"codigoPostal", cp}}
+	cur, err := collection.Find(context.TODO(), filter, findOpts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	for cur.Next(context.TODO()) {
+		// create a value into which the single document can be decoded
+		var s models.ZipCodeCSV
+		err := cur.Decode(&s)
+		if err != nil {
+			log.Fatal(err)
+		}
+		results = append(results, s)
+	}
+	if err := cur.Err(); err != nil {
+		log.Fatal(err)
+	}
+	cur.Close(context.TODO())
+	return results
+}
+
+func saveZipCodes(zip []interface{}, collection *mongo.Collection) *mongo.InsertManyResult {
+	insertManyResult, err := collection.InsertMany(context.TODO(), zip)
+	if err != nil {
+		log.Fatal(err)
+	}
+	return insertManyResult
+}
+
+func getCSVCodes() ([]interface{}, []models.ZipCodeCSV) {
+	getZipCodesCSVFile()
+	csvFile, err := os.Open("zipcodes.csv")
+	if err != nil {
+		log.Fatalln("Couldn't open the csv file", err)
+	}
+	zipCodes, zipCodesModel := csvToMap(csvFile)
+	return zipCodes, zipCodesModel
+}
+
+func getZipCodesCSVFile() {
+	url := "https://www.correosdemexico.gob.mx/datosabiertos/cp/cpdescarga.txt"
+	response, err := http.Get(url)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer response.Body.Close()
+	stream := charmap.ISO8859_1.NewDecoder().Reader(response.Body)
+	csvFile, err := os.Create("zipcodes.csv")
+	if err != nil {
+		log.Fatalf("failed creating file: %s", err)
+	}
+	defer csvFile.Close()
+	_, err = io.Copy(csvFile, stream)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func csvToMap(reader io.Reader) ([]interface{}, []models.ZipCodeCSV) {
+	var zipCodes []interface{}
+	var zipCodesModel []models.ZipCodeCSV
+	r := csv.NewReader(reader)
+	r.Comma = '|'
+	r.LazyQuotes = true
+	r.FieldsPerRecord = -1
+	var header []string
+	firstLine := true
+	for {
+		if firstLine == false {
+			record, err := r.Read()
+			if err == io.EOF {
+				break
+			}
+			if err != nil {
+				log.Fatal(err)
+			}
+			if header == nil {
+				header = record
+			} else {
+				dict := map[string]string{}
+				for i := range header {
+					dict[header[i]] = record[i]
+				}
+				u := strings.Replace(uuid.New().String(), "-", "", -1)
+				isoCode, err := strconv.Atoi(dict["c_estado"])
+				if err != nil {
+					continue
+				}
+				parsingZipCode := models.ZipCodeCSV{
+					Id:           u,
+					CodigoPostal: dict["d_codigo"],
+					Estado:       dict["d_estado"],
+					EstadoISO:    rows[isoCode]["isoCode"],
+					Municipio:    dict["D_mnpio"],
+					Ciudad:       dict["d_ciudad"],
+					Barrio:       dict["d_asenta"],
+				}
+				zipCodes = append(zipCodes, parsingZipCode)
+				zipCodesModel = append(zipCodesModel, parsingZipCode)
+			}
+		} else {
+			r.Read()
+			firstLine = false
+		}
+	}
```

Comment:
I believe this code could be refactored in some way, maybe we can take advantage of a mentoring session to go over this.